contracts/MechMarketplace.sol

```diff
@@ -43,8 +56,11 @@ contract MechMarketplace is IErrorsMarketplace {
     event SetMechFactoryStatuses(address[] mechFactories, bool[] statuses);
     event SetPaymentTypeBalanceTrackers(bytes32[] paymentTypes, address[] balanceTrackers);
     event MarketplaceRequest(address indexed requester, address indexed requestedMech, uint256 requestId, bytes data);
-    event MarketplaceDeliver(address indexed priorityMech, address indexed actualMech, address indexed requester,
+    event MarketplaceDeliver(address indexed priorityMech, address indexed deliveryMech, address indexed requester,
         uint256 requestId, bytes data);
+    event MarketplaceDeliverBatch(address indexed priorityMech, address indexed deliveryMech, address indexed requester,
```

Comment:
Not related to this event: how is a race condition prevented between delivering mechs that have all compete for the same on-chain balance of the requester?

Similarly, how does priority vs delivering mech distinction work, if at all?

Reply:
Ah yes, I keep forgetting to check for requests not being executed on-chain.

Reply:
Here the priority mech is strictly a delivery mech, as it's already delivering requests. The idea is:
- Mech collects signed requests;
- Checks that the user has enough balance to cover delivery (easy to do off-chain by checking user balance);
- Processes requests;
- Provides raw input request data and signatures as proofs;
- Provides corresponding delivery requests data and delivery rates.

To be on a safe side it still makes sense to remove requester ability to withdraw their funds.

Reply:
Talked to Ardian, he also favors ban on withdraws for requesters.